### PR TITLE
Make the AudioDestinationNode have an output

### LINKED ISSUE #349: Changing number of AudioDestinationNode outputs from 0 to 1
_Reported by ROC in http://lists.w3.org/Archives/Public/public-audio/2014JulSep/0145.html_

This came up in public-media-capture. Suppose you want to capture the output of an AudioContext to a MediaStream or MediaRecorder. The natural thing to do would be something like
```
  var d = context.createMediaStreamDestination();
  context.destination.connect(d);
```
or
```
  var r = new MediaRecorder(context.destination);
```
but that doesn't work because the AudioDestinationNode has no outputs.

So how about we make AudioDestinationNode output the AudioContext output (the mix of its inputs)? Then the above code would work. I'm quite confident it would be a Web-compatible change.

### PR DESCRIPTION
**Pull request by padenot**
This fixes #349.

### CODE REVIEW

**Comment by joeberkovitz** [comment]
@cwilso retracting objection, so this is ready to merge: there is a `maxChannelCount` attribute unique to `AudioDestinationNode`

**Comment by cwilso** [comment]
After careful review of the original issue and Paul's PR, I retract my comment.  LGTM.

File: index.html [hunk 1]
```diff
@@ -1182,15 +1178,25 @@ <h3 id="lifetime-AudioNode" class="informative">Lifetime</h3>
...
+  rendered audio to be heard will be routed to this node, a "terminal" node in
+  the <a><code>AudioContext</code></a>'s routing graph. There is only a single
+  AudioDestinationNode per <a><code>AudioContext</code></a>, provided through
+  the <code>destination</code> attribute of <a><code>AudioContext</code></a>.
+</p>
+<p>
+  The output of a <a><code>AudioDestinationNode</code></a> is the sum of its
+  inputs, allowing to capture the output of an <a><code>AudioContext</code></a>
+  into, for example, a <a><code>MediaStreamAudioDestinationNode</code></a>, or a
+  <a><code>MediaRecorder</code></a> (described in <cite> <a href="http://XXX">
```
Comment (rtoy) [inline index.html]: Fix the link.

**Comment by rtoy** [comment]
Except for the link, I think this is ready to go, once the conflicts are resolved.

**Comment by padenot** [comment]
This is now fixed, sorry for the delay.

**Comment by padenot** [comment]
Rebased and tidied.

**Comment by padenot** [comment]
Rebased, tidied, link added.

**Comment by rtoy** [comment]
Sorry for the long delay!  I missed this.

**Pull request merged**